adds new junos_facts module

privateip opened the pull request:
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

network/junos/junos_facts
##### ANSIBLE VERSION

```
ansible 2.1.0 (local_action_junos_config edbdf0d955) last updated 2016/04/22 15:49:41 (GMT -400)
  lib/ansible/modules/core: (working b1ab9b93b7) last updated 2016/04/22 15:37:32 (GMT -400)
  lib/ansible/modules/extras: (working 6962f8af62) last updated 2016/03/27 22:50:50 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

The junos_facts module will collect basic system inforamtion about remote
devices running the Junos operating system

Commit:
adds new junos_facts module

The junos_facts module will collect basic system inforamtion about remote
devices running the Junos operating system

gregdek commented:
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

privateip committed:
Merge pull request #9 from dgarros/junos_modules

Add support for xml, set and text format for config

Commit:
adds config_format argument to the facts module

this change adds config_format argument with choices of xml, set or text to
specify the desired format of the config returned from the remote device.  The
default value is text
